### PR TITLE
cmd/enable: migrate old data dir on enable

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Adding code to migrate /var/lib/azure/custom-script to
/var/lib/waagent/custom-script so that it gets cleaned up on
'waagent -deprovision' path. Shelling out to `mv -f` for recursive
file copying.

Also bumped version to v2.0.1.

cc: @boumenot

### CODE REVIEW

File: integration-test/test/handler-commands.bats [hunk 1]
```diff
@@ -98,10 +98,26 @@ teardown(){
...
+		touch \$OLD/sentinel.txt && \
+		fake-waagent install && \
+		fake-waagent enable && wait-for-enable"
+    # download an external script and run it
+    push_settings '{"commandToExecute":"date"}'
+    run start_container
+    echo "$output"
+
+    diff="$(container_diff)"; echo "$diff"
+    [[ "$diff" != *"A /var/lib/azure/custom-script"* ]] # should have gone
```
Comment (boumenot) [inline integration-test/test/handler-commands.bats]: NIT: *be* gone

File: main/cmds.go [hunk 2]
```diff
@@ -61,6 +61,13 @@ func uninstall(ctx *log.Context, h vmextension.HandlerEnvironment, seqNum int) e
 }
 
 func enable(ctx *log.Context, h vmextension.HandlerEnvironment, seqNum int) error {
+	// for a few versions we need to migrate dataDirOld (introduced in v2.0.0) to
+	// dataDir (introduced in v2.0.1).
+	ctx.Log("message", "checking for state migration")
```
Comment (boumenot) [inline main/cmds.go]: The nice thing is that this is a patch version, so once 2.0.1 is everywhere we no longer have to worry about this.

File: main/migration.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+)
+
+// migrateDataDir moves oldDir to newDir, if oldDir exists by shelling out to
+// 'mv -rf'.
```
Comment (boumenot) [inline main/migration.go]: NIT: mv -f is what you have below.

**Comment by boumenot** [comment]
LGTM.

**Comment by ahmetalpbalkan** [comment]
@boumenot release?

**Pull request merged**